File: backend-springboot/src/main/java/com/backend/spring/modules/usercontext/user/services/UserService.java
package com.backend.spring.modules.usercontext.user.services;

import com.backend.spring.modules.usercontext.user.entities.User;

import java.util.List;
import java.util.Optional;

public interface UserService {
    public Optional<User> findByEmail(String email);
    public User save(User user);
    public User signup(User user);
    public User updateUser(User newUser, User oldUser);
    public List<User> findAll();
    public Optional<User> findById(Long id);
    public void deleteById(Long id);
    public void delete(User user);
    public User updateUserFromAdmin(User newUser, User oldUser);
}

File: backend-nodejs-ts/src/modules/user/services/UserService.ts
import { User, UserModel } from '../models/UserModel';

class UserService {
  public findByEmail(email: string) {
    const user = UserModel.findOne({ email });
    return user;
  }

  public save(user: User) {
    const newUser = UserModel.create(user);
    return newUser;
  }

  public findById(id: string) {
    const user = UserModel.findById(id).select('-password');
    return user;
  }

  public findAll() {
    const users = UserModel.find({}).select('-password');
    return users;
  }

  public async deleteUser(userId: string) {
    const user = UserModel.findById(userId);

    if (user) {
      await user.remove();
      return true;
    }

    return false;
  }
}

export const userService = new UserService();
